### PR TITLE
Squish shitcode and a future bug

### PR DESCRIPTION
**Pull request by Lumipharon**
## About The Pull Request
Projectile code still had a funny bit of code where it did a return prob(.). This works fine in 514.1584, but in 1585 this breaks entirely, meaning it would be impossible to shoot past objects.

Fixes this bug before it becomes an actual problem, and clean up the shitcode.

## Changelog
:cl:
fix: fixes a projectile bug for a future version of byond
/:cl:

### CODE REVIEW

File: code/modules/projectiles/projectile.dm [hunk 1]
```diff
@@ -656,7 +656,7 @@ So if we are on the 32th absolute pixel coordinate we are on tile 1, but if we a
 		return TRUE
 	if(proj.distance_travelled <= proj.ammo.barricade_clear_distance)
 		return FALSE
-	. = coverage //Hitchance.
+	var/hit_chance = coverage //base chance for the projectile to hit the object instead of bypassing it
```
Comment (BraveMole) [inline code/modules/projectiles/projectile.dm:659]: why do you even need a new var when coverage does the trick

Comment (BraveMole) [reply code/modules/projectiles/projectile.dm:659]: Also there should be no difference between . and a "normal" var.

If there is, you really need to report that to lummox

**Review by BraveMole — CHANGES_REQUESTED** [review]

File: code/modules/projectiles/projectile.dm (hunk 1 above)
Comment (Lumipharon) [reply code/modules/projectiles/projectile.dm:659]: I bugreported it.

The var is modified before being returned. If I use coverage directly that would be permanently changing the coverage var of the object.

Comment (Lumipharon) [reply code/modules/projectiles/projectile.dm:659]: Oh he actually responded. 
image

**Pull request merged**